web/src/components/FilteredConnection.tsx

```diff
@@ -466,18 +538,37 @@ export class FilteredConnection<N, NP = {}, C extends Connection<N> = Connection
 
                         const result = this.props
                             .queryConnection({
-                                first: this.state.first,
+                                first: this.state.first + ((!this.state.queried && this.state.previousPagesCount) || 0),
+                                after: this.state.after,
                                 query,
                                 ...(filter ? filter.args : {}),
                             })
                             .pipe(
                                 catchError(error => [asError(error)]),
-                                map(
-                                    (c): PartialStateUpdate => ({
-                                        connectionOrError: c,
-                                        connectionQuery: query,
-                                        loading: false,
-                                    })
+                                tap(() => this.setState({ queried: true })),
+                                mergeMap(
+                                    async (c: C | ErrorLike): Promise<PartialStateUpdate> => {
+                                        // If we're paging with cursors and we have a new set of results,
+                                        // append them onto the currently visible result set and return
+                                        // the new concatenated set of results.
+                                        if (c && !isErrorLike(c) && this.props.cursorPaging) {
+                                            await new Promise(resolve => {
+                                                this.setState(
+                                                    state => ({ previousPage: state.previousPage.concat(c.nodes) }),
+                                                    () => {
+                                                        c.nodes = this.state.previousPage
```

Comment:
Just because this wasn't completely obvious to me at first, I think a comment like this could be helpful:

```suggestion
                                                        // Update the connection's nodes to the concatenated set of results.
                                                        c.nodes = this.state.previousPage 0 
```